File: diagonalDifference.js
//DIAGONAL DIFFERENCE;

/*      [1,2,3]
        [4,5,6]
        [9,8,9]
*/

let n = 3;
let arr = [[1, 2, 3], [4, 5, 6], [9, 8, 9]];


function diagonalDifference(arr) {
        sumOne = 0
        sumTwo = 0
        for (let i = 0; i < arr.length; i++) {
                sumOne += (arr[i][i])
        }
        for (let j = 0; j < arr.length; j++) {
                sumTwo += (arr[j][(arr.length - 1) - j])
        }
        return Math.abs(sumOne - sumTwo)
}

console.log(diagonalDifference(arr))

File: diagonal_difference.py
n = 3
arr = [[1, 2, 3], [4, 5, 6], [9, 8, 9]]


def diagonalDifference(arr):
    sum_one = 0
    sum_two = 0
    for i in range(n):
        sum_one += (arr[i][i])
    for j in range(n):
        sum_two += (arr[j][(n-1) - j])
    return(abs(sum_one - sum_two))


print(diagonalDifference(arr))
